Library/Homebrew/utils/tty.rb

```diff
@@ -0,0 +1,51 @@
+module Tty
+  module_function
+
+  def strip_ansi(string)
+    string.gsub(/\033\[\d+(;\d+)*m/, "")
+  end
+
+  def width
+    `/usr/bin/tput cols`.strip.to_i
+  end
+
+  def truncate(string)
+    (w = width).zero? ? string.to_s : string.to_s[0, w - 4]
+  end
+
+  COLOR_CODES = {
+                  red:     31,
+                  green:   32,
+                  yellow:  33,
+                  blue:    34,
+                  magenta: 35,
+                  cyan:    36,
+                  default: 39,
+                }
+
+  STYLE_CODES = {
+                  reset:          0,
+                  bold:           1,
+                  italic:         3,
+                  underline:      4,
+                  strikethrough:  9,
+                  no_underline:  24,
+                }
+
+  CODES = COLOR_CODES.merge(STYLE_CODES)
+
+  CODES.each do |name, code|
+    define_singleton_method(name) do
+      @sequence = [*@sequence, code]
```

Comment:
I don't know, really. I just find the current implementation confusing; it's not obvious that it's for changing and how the `@sequence` stuff works.

Some vague thoughts:
```ruby
@sequence ||= []
@sequence << code
self
```
signals intent a bit more. Even then, I'd be tempted to make a module-level method that is called `append_code` that encapsulates this logic so the `define_singleton_method` just ends up calling it. Similarly with `[*@sequence].empty?` below; having that be a separate module-level method would be good. Finally, using an attribute reader for `sequence` seems better than having raw access given how it's intended to be used.